Azure/azure-cli
# [RDBMS] Breaking Change: Parameter name updates. Improvements to Management Plane for MySQL and PostgreSQL

**Description**  

* Note: Recreating this PR to update the history/rebase with hotfix changes for rdbms.

This PR improves existing commands for the 'Flexible Servers' APIs within the RDBMS component.  

Breaking changes:
_Parameter name changes_
1. -- time to --restore-in-time for restore command
Why: PM requested change to make it clearer for customer.  Documentation on our side will also be updated. 
2. --force to --yes 
Why: Consistency across different delete prompt commands for customer experience.
3. --server-name to --name for CRUD operations
Why: PM requested change for consistency
4. --name to --rule-name in firewall rule
Why: PM requested for clearer & self-explanatory params for customer experience.

Changes include:

- Update MySQL Connection string in create command
- Replace server name check with CheckNameAvailability API for MySQL & PostgreSQL
- Update help text and parameters for commands to be more descriptive and helpful to user (commands/arguments: vnet related, replica, firewall, restore).  These were to improve customer experience.
- Update validators to use result from list-skus commands/API.  
- Enabled more tests for code coverage and command testing (validator & replica tests)
- Added new recording files with default locations for MySQL & PostgreSQL 

**Testing Guide**  


To run tests (these cover existing commands and include new tests):
azdev test rdbms

---

This checklist is used to make sure that common guidelines for a pull request are followed.
